src/factories/Axis.ts

```diff
@@ -30,13 +32,73 @@ module n3Charts.Factory {
       this.scale = this.getScale(dim);
       this.axis = this.getAxis();
 
-      this.updateAxis(dim);
+      this.updateScaleDomain(datasets, options);
+      this.updateAxisSize(dim);
     }
 
     destroy() {
       this.destroyAxis();
     }
 
+    updateScaleDomain(datasets, options) {
+      this.scale.domain(this.getExtent(datasets, options));
+    }
+
+    getExtentForDatasets(
+      datasets,
+      filter: (key:string) => Boolean,
+      accessor: (datum, datasetKey:string) => number[]
+    ) {
+      var min = Number.POSITIVE_INFINITY;
+      var max = Number.NEGATIVE_INFINITY;
+
+      for (var key in datasets) {
+        if (!filter(key)) continue;
+
+        datasets[key].forEach((datum) => {
+          var data = accessor(datum, key);
+          if (data[0] < min) min = data[0];
+          if (data[1] > max) max = data[1];
+        });
+      }
+
+      return [min, max];
+    }
+
+    getExtent(datasets, options) {
+      if (this.isAbscissas()) {
+        var abscissasKey = options.axes.x.key;
+        return this.getExtentForDatasets(
+          datasets,
+          () => true,
+          (datum) => [datum[abscissasKey], datum[abscissasKey]]
+        );
+      }
+
+      var datasetsForSide = [];
+      var seriesForDataset = {};
+      options.series.forEach((series) => {
+        if (series.axis === this.side) {
+          datasetsForSide.push(series.dataset);
+          if (!seriesForDataset[series.dataset]) seriesForDataset[series.dataset] = [];
+          seriesForDataset[series.dataset].push(series);
+        }
+      });
+
+      return this.getExtentForDatasets(
+        datasets,
+        (key) => datasetsForSide.indexOf(key) > -1,
```

Comment:
can we make this a little easier? i mean, i can follow - but I need to jump to the `getExtentForDatasets` method to understand what this is doing. We can still merge this and I can debug a little to propose a simplification.